app/src/main/java/nextstep/signup/ui/SingUpScreen.kt

```diff
@@ -0,0 +1,159 @@
+package nextstep.signup.ui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import nextstep.signup.ui.theme.Blue50
+
+@Preview(showBackground = true)
+@Composable
+fun SignUpScreen() {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = 32.dp)
+    ) {
+        SignUpTitle()
+        Spacer(modifier = Modifier.height(height = 42.dp))
+        UserNameTextField()
+        Spacer(modifier = Modifier.height(height = 36.dp))
+        EmailTextField()
+        Spacer(modifier = Modifier.height(height = 36.dp))
+        PasswordTextField()
+        Spacer(modifier = Modifier.height(height = 36.dp))
+        PasswordConfirmTextField()
+        Spacer(modifier = Modifier.height(height = 42.dp))
+        SignUpButton()
```

Comment:
Compose API Guidelines서는 모든 컴포저블 함수에 modifier를 첫 번째 선택적 매개변수(named arguments 활용)로 받을 수 있도록 구현해달라고 합니다.

가이드에 설명된 내용을 보면 modifier를 매개변수로 받지 않는다면, 해당 컴포저블 함수에 추가로 UI 요구사항이 있을 때 또 다른 컴포저블 함수로 감싸 modifier를 받아야 하는 등의 경우를 방지하기 위해서라고 설명합니다.
https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-api-guidelines.md#elements-accept-and-respect-a-modifier-parameter

이렇게 modifier를 파라미터에 추가한다면 꼭 각 컴포넌트의 공백을 줄 때 매번 Spacer를 활용하지 않아도 될 것 같아요~!

Reply:
또는 이번 기회에 Arrangment.spacedBy에 대해 학습하고 적용해보시면 어떨까요?

Reply:
> 이렇게 modifier를 파라미터에 추가한다면 꼭 각 컴포넌트의 공백을 줄 때 매번 Spacer를 활용하지 않아도 될 것 같아요~!

Spacer를 활용하지 않아도 된다는 말씀은, padding을 활용하는 것을 의도하신 것 같은데요.
수현님은 어떤 경우에 Spacer를 쓰고 어떤 경우에 padding을 사용하시나요?

> 또는 이번 기회에 Arrangment.spacedBy에 대해 학습하고 적용해보시면 어떨까요?

Arrangement.spaceBy를 활용하면 어떤 이점이 있을까요?

Reply:
@BeokBeok 
> Spacer를 활용하지 않아도 된다는 말씀은, padding을 활용하는 것을 의도하신 것 같은데요.
수현님은 어떤 경우에 Spacer를 쓰고 어떤 경우에 padding을 사용하시나요?

저는 Compose API Guidelines에서 가이드하듯이 modifier를 넘긴 형태로 구현한 경우에는 굳이 Spacer를 사용하지 않고 padding을 활용했었는데요,

현석님이 남겨주신 코멘트를 보고 각각 어떤 상황에서 사용하는 것이 적절할지 찾아보다 흥미로운 글을 발견했어요!
https://www.jetpackcompose.app/articles/space-hoisting-should-I-use-the-padding-modifier-or-the-spacer-composable#space-hoisting

> A lot of components react to gestures and animations. Imagine a row component that shows the ripple effect when you tap on it. If we were to pass in the padding Modifier to that component, even this extra space that we added would react to the gesture. As the order of the Modifiers matter, you might be able to avoid this problem, but it would depend on the implementation. Using Spacer simplifies this and avoids the problem altogether.

예를 들어, 탭할 때 리플 효과를 보여주는 Row 컴포넌트에 패딩 Modifier를 적용하면 리플 효과가 예상치 못하게 적용될 수 있습니다. Modifier의 순서를 잘 조절하면 구현에 따라 이 문제를 피할 수 있지만, Spacer를 사용하면 이 문제를 간단히 해결하고 완전히 피할 수 있다고 설명하네요! 덕분에 저도 새로운 시야를 넓힐 수 있었습니다 👍 

> Arrangement.spaceBy를 활용하면 어떤 이점이 있을까요?

Arrangement.spaceBy를 활용하면 요소들 간 동일한 공백을 갖는 경우 해당 요소들을 묶어 동일하게 공백을 줄 수 있습니다.
나중에 비슷한 요소들 간 공백이 동일하게 수정되더라도 변경사항을 줄일 수 있습니다.

Reply:
> Arrangement.spaceBy를 활용하면 요소들 간 동일한 공백을 갖는 경우 해당 요소들을 묶어 동일하게 공백을 줄 수 있습니다.
나중에 비슷한 요소들 간 공백이 동일하게 수정되더라도 변경사항을 줄일 수 있습니다.

말씀하신 부분에 대해서 동의합니다.
다만, 제공해주신 피그마 가이드를 보시면 TextField간에는 동일한 여백(36)을 가지지만 타이틀과 버튼과의 여백(42)은 다른데요.
spaceBy를 활용하기 위해 별도의 TextField들을 감싸는 Column을 만들어야 하는 것일까요?